browser/brave_shields/ad_block_subscription_download_manager_getter.cc

```diff
@@ -80,26 +80,34 @@ class AdBlockSubscriptionDownloadManagerGetterImpl
         profile_manager->user_data_dir().Append(
             profile_manager->GetInitialProfileDir()));
     if (profile) {
-      GetDownloadManager(profile);
+      base::ThreadTaskRunnerHandle::Get()->PostTask(
+          FROM_HERE,
+          base::BindOnce(
+              &AdBlockSubscriptionDownloadManagerGetterImpl::GetDownloadManager,
+              base::Unretained(this), profile->GetProfileKey()));
```

Comment:
ProfileKey is a part of a profile, so when profile dies, the key dies as well

Reply:
fixed with some overall reorganization thanks to @bridiver